src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -457,20 +505,36 @@ public void finishTree(DetailAST rootAST) {
             return;
         }
 
-        final ImportDetails firstImport = importToGroupList.get(0);
-        String currentGroup = getImportGroup(firstImport.isStaticImport(),
-                firstImport.getImportFullPath());
+        String currentGroup = getNextImportGroup(0);
         int currentGroupNumber = customImportOrderRules.indexOf(currentGroup);
         String previousImportFromCurrentGroup = null;
+        
+        ImportDetails firstImportObject = importToGroupList.get(0);
+        final String firstImportGroup = firstImportObject.getImportGroup();
+        final String firstImportFullIdent = firstImportObject.getImportFullPath();
+        if( firstImportGroup == NON_GROUP_RULE_GROUP ) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/checkstyle/checkstyle/file/1008138044/issues/source?bid=2193633&fileBranchId=2722517#l515)